### ISSUE TITLE
`@cue` needs clarification

### ISSUE THREAD

**Issue by th-we** [music-encoding/music-encoding#668]
The [documentation for `@cue`](https://music-encoding.org/guidelines/v4/attribute-classes/att.cue.html) just says

> Attributes that describe "cue-ness".

Can we define what cue-ness means?  Specifically, does this mean that a note is (logically) a cue from a different part, or does it mean that the note is (visually) cue size?  There are some contradictions here: 

* `@cue` is a logical domain attribute for [notes](https://music-encoding.org/guidelines/v4/attribute-classes/att.note.log.html), [chords](https://music-encoding.org/guidelines/v4/attribute-classes/att.chord.log.html), [rests](https://music-encoding.org/guidelines/v4/attribute-classes/att.rest.log.html) and [measure rests](https://music-encoding.org/guidelines/v4/attribute-classes/att.mrest.log.html).
* But it's also a visual domain attribute for [beams](https://music-encoding.org/guidelines/v4/attribute-classes/att.note.log.html) and [beamSpans](https://music-encoding.org/guidelines/v4/attribute-classes/att.beamspan.vis.html).

It should not be used for both those domains. While the original intent seems to be flagging logical cues (#433, #406), the actual usage seems to be focused on visual size:

* Verovio [imports](https://github.com/rism-ch/verovio/blob/d746305f29483d14dec8bd5a300ef9f7050b3d58/src/iomusxml.cpp#L2173) MusicXML's "logical" `<cue>` element the same way as the size-related ("visual") `type/@size='cue'` attribute instead of converting one to `@cue='true'` and the other to `@fontsize='small'`
* The only occurrence of [`@cue` in the sample encodings](https://github.com/music-encoding/sample-encodings/blob/master/MEI_4.0/Music/Complete_examples/Parker-Gillespie_ShawNuff.mei#L209) seems to use it for size (there is no part that the small notes "quote", it's just more music that is added to the leadsheet)
* `@cue` was suggested for [encoding small "cadenza-style"/rhythmically free-floating notes](https://github.com/music-encoding/music-encoding/issues/509)
* [`@cue` was added to beams](https://github.com/music-encoding/music-encoding/commit/8507516acfe8b07b37eb15e7de9721b1781e74c8), motivated by cases where [regular-sized and cue-sized notes share a beam](https://github.com/music-encoding/music-encoding/issues/558)

As we already have `@fontsize` as a visual domain attribute, I plead `@cue` should be reserved for actual logical cues only, despite its current de-facto usage. (Whether `@fontsize` would be a sensible attribute for `<beam>` is to be discussed.) We don't have any other way of identifying logical cues.

(If we really want an encoding equivalent of the established term "cue size" for any notation elements that are small, we could extend [data.FONTSIZETERM](https://music-encoding.org/guidelines/v4/data-types/data.fontsizeterm.html) and add a `'cue'` value. But I think `@fontsize='small'` is just fine and we could document that this is the preferred value for anything "cue size".)

**Comment by ahankinson** [comment]
I've been thinking about this one in relation to #859.

> Specifically, does this mean that a note is (logically) a cue from a different part, or does it mean that the note is (visually) cue size? 

My understanding of the musical domains is not that something (an attribute or element) belongs in one or another _exclusively_. Many aspects of music notation are a variable combination of the visual / gestural / logical domains, rather than clearly in one domain or another, so saying something is in one domain or another exclusively is to deny the fact that musical symbols exist in all domains simultaneously.

Rather, the domains are in MEI to explicitly acknowledge the fact that symbols operate in several different ways simultaneously, and so that you can *address* certain features of domain in a musical element, to change what otherwise might be expected. 

In this way, I don't think we really need to clarify the domains of 'cue-ness'. A cue is a cue. **It is in all domains at the same time**. You can use `@fontsize` to change the size of the cue, if you like. 

Yes, the docs can always improved, but I think that improving them by making an fixed boundary between visual and logical is not the right way to approach it.

**Comment by th-we** [comment]
@ahankinson I'm perfectly O.K. with considering `@cue` belonging to both the logical and visual domains. But that means that it has implications for both domains, no?  I.e. it implies that the element belongs to a cue and that it is rendered at cue size (unless `@fontsize` overrides this).

The examples I listed only imply that elements are smaller, but not cues.  I suggest that something like `@fontsize` is used in those cases where we don't deal with cues.

**Comment by ahankinson** [comment]
Sure -- but is this a modification for the MEI spec, or a modification to the examples you posted?

**Comment by th-we** [comment]
Yes, the examples should be changed, but as `@cue` seems to be misused so frequently, I think it also needs some clarification in the documentation about what it is and what it is not.

**Comment by kepper** [comment]
related to #926

**Comment by bwbohl** [comment]
cf #926

**Comment by bwbohl** [comment]
Pushing this to MEI 5.x milestone because the associated PR is still a draft and discussion points to additional thought that has to be put into this matter.
